pkg/hakeeper/rsm_test.go

```diff
@@ -339,3 +340,89 @@ func TestScheduleCommandQuery(t *testing.T) {
 	}
 	assert.Equal(t, b, *cb)
 }
+
+func TestInitialState(t *testing.T) {
+	rsm := NewStateMachine(0, 1).(*stateMachine)
+	assert.Equal(t, pb.HAKeeperCreated, rsm.state.State)
+}
+
+func TestSetState(t *testing.T) {
+	tests := []struct {
+		initialState pb.HAKeeperState
+		newState     pb.HAKeeperState
+		result       pb.HAKeeperState
+	}{
+		{pb.HAKeeperCreated, pb.HAKeeperBootstrapping, pb.HAKeeperCreated},
+		{pb.HAKeeperCreated, pb.HAKeeperBootstrapFailed, pb.HAKeeperCreated},
+		{pb.HAKeeperCreated, pb.HAKeeperRunning, pb.HAKeeperCreated},
+		{pb.HAKeeperCreated, pb.HAKeeperCreated, pb.HAKeeperCreated},
+		{pb.HAKeeperBootstrapping, pb.HAKeeperCreated, pb.HAKeeperBootstrapping},
+		{pb.HAKeeperBootstrapping, pb.HAKeeperBootstrapFailed, pb.HAKeeperBootstrapFailed},
+		{pb.HAKeeperBootstrapping, pb.HAKeeperRunning, pb.HAKeeperRunning},
+		{pb.HAKeeperBootstrapping, pb.HAKeeperBootstrapping, pb.HAKeeperBootstrapping},
+		{pb.HAKeeperBootstrapFailed, pb.HAKeeperBootstrapFailed, pb.HAKeeperBootstrapFailed},
+		{pb.HAKeeperBootstrapFailed, pb.HAKeeperCreated, pb.HAKeeperBootstrapFailed},
+		{pb.HAKeeperBootstrapFailed, pb.HAKeeperBootstrapping, pb.HAKeeperBootstrapFailed},
+		{pb.HAKeeperBootstrapFailed, pb.HAKeeperRunning, pb.HAKeeperBootstrapFailed},
+		{pb.HAKeeperRunning, pb.HAKeeperRunning, pb.HAKeeperRunning},
+		{pb.HAKeeperRunning, pb.HAKeeperCreated, pb.HAKeeperRunning},
+		{pb.HAKeeperRunning, pb.HAKeeperBootstrapping, pb.HAKeeperRunning},
+		{pb.HAKeeperRunning, pb.HAKeeperBootstrapFailed, pb.HAKeeperRunning},
+	}
+
+	for _, tt := range tests {
+		rsm := stateMachine{
+			state: pb.HAKeeperRSMState{
+				State: tt.initialState,
+			},
+		}
+		cmd := GetSetStateCmd(tt.newState)
+		rsm.Update(sm.Entry{Cmd: cmd})
```

Comment:
Static-check Error: Error return value of `rsm.Update` is not checked (errcheck)